fhir-server/src/main/java/com/ibm/fhir/server/resources/FHIRResource.java

```diff
@@ -419,12 +422,17 @@ protected String getRequestBaseUri(String type) throws Exception {
 
         // Strip off any path elements after the base
         if (type != null && !type.isEmpty()) {
-            int resourceNamePathLocation = baseUri.lastIndexOf("/" + type);
+            int resourceNamePathLocation = baseUri.indexOf("/" + type + "/");
```

Comment:
I think you want to use lastIndexOf here, so if you did happen to have the case where a path segment matched a particular resource type exactly, but the resource type was specified at the end, you'd still parse correctly (e.g. `https://example.com/my/Patient/api/Patient/123`),

Reply:
the problem with lastIndexOf here is if a user created a resource with id "Patient" that logic would break.
for example:  `https://example.com/fhir/Patient/Patient/_history`

i'd rather err on the side of "operators should do the right thing" rather than "users should do the right thing".

Reply:
I'm ok with that. Should probably document it if there's somewhere that it makes sense to do so.